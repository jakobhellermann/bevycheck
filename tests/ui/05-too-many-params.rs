#![allow(unused)]
use bevy::prelude::*;

struct Foo;

#[bevycheck::system]
fn system(
    _: Res<Foo>,
    _: Res<Foo>,
    _: Res<Foo>,
    _: Res<Foo>,
    _: Res<Foo>,
    _: Res<Foo>,
    _: Res<Foo>,
    _: Res<Foo>,
    _: Res<Foo>,
    _: Res<Foo>,
    _: Res<Foo>,
    _: Res<Foo>,
    _: Res<Foo>,
    _: Res<Foo>,
    _: Res<Foo>,
    _: Res<Foo>,
    _: Res<Foo>,
) {
}

fn main() {
    IntoSystem::into_system(system);
}
