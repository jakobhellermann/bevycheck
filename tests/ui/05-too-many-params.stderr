error: too many system parameters
  --> $DIR/05-too-many-params.rs:7:1
   |
7  | / fn system(
8  | |     _: Res<Foo>,
9  | |     _: Res<Foo>,
10 | |     _: Res<Foo>,
...  |
24 | |     _: Res<Foo>,
25 | | ) {
   | |_^
   |
   = note: only up to 16 parameters are supported
   = help: try bundling some parameters into tuples or a `#[derive(SystemParam)]` struct
